### PR TITLE
Adding umpire_util dependency to umpire_device lib

### PR DESCRIPTION
**Pull request by kab163**
In order to allow error handling for the DeviceAllocator, we need to add the umpire_util dependency to the umpire_device lib.  There were some other additional edits that were made so that error handling in the DeviceAllocator files were correct. We also added "+device_alloc" to a lassen CI job so that the DeviceAllocator is tested in the CI and these issues can be found sooner.

### CODE REVIEW

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
Can you add a GitLab job that tests this? (Or update an existing job)

**Comment by kab163** [comment]
> Can you add a GitLab job that tests this? (Or update an existing job)

Good idea. Let's hope I did that right! Lol
Toda!! https://lc.llnl.gov/gitlab/radiuss/Umpire/-/jobs/622319

**Review by davidbeckingsale — APPROVED** [review]

File: src/umpire/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,7 @@ if (UMPIRE_ENABLE_CUDA AND UMPIRE_ENABLE_DEVICE_ALLOCATOR)
                               PROPERTIES
                               LANGUAGE CUDA)
 
-  set(umpire_device_depends camp cuda)
+  set(umpire_device_depends camp cuda umpire_util)
```
Comment (mcfadden8) [inline src/umpire/CMakeLists.txt:52]: Why `umpire_util`?  The helper seems to be calling `umpire::fmt::format`, not util.  Shouldn't this be:
```suggestion
  set(umpire_device_depends camp cuda umpire_tpl_fmt)
```

Comment (kab163) [reply src/umpire/CMakeLists.txt:52]: I just tested that and it works with that as well. @davidbeckingsale should I change it to `umpire_tpl_fmt` instead?

Comment (davidbeckingsale) [reply src/umpire/CMakeLists.txt:52]: The error classes are provided under umpire_util so I think that is what should be used. The fmt dependency will get brought in transitively.

**Pull request merged**